cds_ils/importer/documents/importer.py

```diff
@@ -0,0 +1,110 @@
+import uuid
+from copy import deepcopy
+
+import click
+from invenio_app_ils.documents.api import DocumentIdProvider
+from invenio_app_ils.errors import IlsValidationError
+from invenio_app_ils.proxies import current_app_ils
+from invenio_db import db
+
+from cds_ils.importer.documents.api import search_documents_by_isbn, \
+    search_documents_by_doi, get_document_by_exact_title_authors
+
+
+class DocumentImporter(object):
+
+    def __init__(self, json_metadata, helper_metadata_fields,
+                 metadata_provider, update_document_fields):
+        self.helper_metadata_fields = helper_metadata_fields
+        self.json_data = json_metadata
+        self.metadata_provider = metadata_provider
+        self.update_document_fields = update_document_fields
+
+    def _set_record_import_source(self, record_dict):
+        record_dict["created_by"] = {"type": "import",
+                                     "value": self.metadata_provider}
+
+    def _before_create(self):
+        cleaned = deepcopy(self.json_data)
+        for field in self.helper_metadata_fields:
+            if field in cleaned:
+                del cleaned[field]
+        # save the import source
+        self._set_record_import_source(cleaned)
+        return cleaned
+
+    def create_document(self):
+        """Create a new record from dump."""
+        cleaned_json = self._before_create()
+        document_class = current_app_ils.document_record_cls
+        # Reserve record identifier, create record and recid pid in one
+        # operation.
+        record_uuid = uuid.uuid4()
+        provider = DocumentIdProvider.create(
```

Comment:
When syncing with CDS (not migration), we might what to store somewhere the CDS recid (which is not the `legacy_id`) to show a button `See this in CDS` or something, WDYT?
Actually, it might be already done in `alternative_identifiers`? Would be great to have a test for this.

Reply:
I thought it would be the `legacy_recid`. Did you have something else in mind?

Reply:
Probably more `alternative_identifiers` instead of adding a new field. We might need to update the vocabulary.

Reply:
but we already have the `legacy_recid` field for the migration, I'm not sure if I understand what do you mean